akka-projection-core/src/main/scala/akka/projection/scaladsl/SourceProvider.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl
+
+import akka.stream.scaladsl.Source
+
+trait SourceProvider[Envelope, Payload, Offset] {
```

Comment:
I'm looking for alternatives to this name. In principle, a `Provider` should work as a factory for something, in this case to the `Source`. 

However, because something needs to know how to extract the parts (`Payload` and `Offset`) it's natural that the one providing it carries that logic.